NAudio/Wave/WaveOutputs/WasapiOut.cs

```diff
@@ -325,6 +325,25 @@ public void Init(IWaveProvider waveProvider)
                             correctSampleRateFormat = null;
                         }
 
+                        // If still null and we're using exclusive mode, try to get the device format property.
+                        if (correctSampleRateFormat == null && shareMode == AudioClientShareMode.Exclusive)
+                        {
+                            // Based on https://stackoverflow.com/questions/22616924/wasapi-choosing-a-wave-format-for-exclusive-output
+                            byte[] waveFormatBytes = (byte[])mmDevice.Properties[PropertyKeys.PKEY_AudioEngine_DeviceFormat].Value;
+                            if (waveFormatBytes != null)

```

Comment:
Well, actually I'm confused.  https://msdn.microsoft.com/en-us/library/windows/desktop/dd316580(v=vs.85).aspx says it's a WAVEFORMATEX, but the dwChannelMask is getting set correctly, which I think means (in my case at least) it must be a WAVEFORMATEXTENSIBLE.

Maybe I need to take a look at the byte count to decide or something.  Let me know if you have any ideas. :-)

To be clear, the code does work for me as-is. *shrug*